modules/apps/headless/headless-asset-library/headless-asset-library-impl/src/main/java/com/liferay/headless/asset/library/internal/resource/v1_0/AssetLibraryResourceImpl.java

```diff
@@ -110,6 +140,44 @@ public Page<AssetLibrary> getAssetLibrariesPage(
 					GetterUtil.getLong(document.get(Field.ENTRY_CLASS_PK)))));
 	}
 
+	@Override
+	public Page<AssetLibrary> getAssetLibrariesPinnedByMePage(
+			Pagination pagination)
+		throws Exception {
+
+		if (!FeatureFlagManagerUtil.isEnabled("LPD-17564")) {
+			throw new UnsupportedOperationException();
+		}
+
+		List<DepotEntryPin> userDepotEntryPins =
+			_depotEntryPinService.getUserDepotEntryPins(
+				contextUser.getUserId(), pagination.getStartPosition(),
+				pagination.getEndPosition());
+
+		List<AssetLibrary> assetLibraries = new ArrayList<>(
+			pagination.getPageSize());
+
+		for (DepotEntryPin depotEntryPin : userDepotEntryPins) {
+			DepotEntry depotEntry = _depotEntryService.getDepotEntry(
+				depotEntryPin.getDepotEntryId());
+
+			if (depotEntry == null) {
+				continue;
+			}
+
+			assetLibraries.add(_toAssetLibrary(depotEntry));
+
+			if (assetLibraries.size() == pagination.getPageSize()) {
+				break;
+			}
```

Comment:
Shouldn't this condition be handled internally by `userDepotEntryPins`? It should be safe for callers to assume that the number of elements returned will always be <= than end - start.